LiteDB/Client/Mapper/BsonMapper.cs

```diff
@@ -30,7 +30,12 @@ public partial class BsonMapper
         /// Mapping cache between Class/BsonDocument
         /// </summary>
         private readonly Dictionary<Type, EntityMapper> _entities = new Dictionary<Type, EntityMapper>();
```

Comment:
I'm not sure it's a good idea to allow concurrent access to `Dictionary`, given that [it's not thread-safe](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0#thread-safety) and there is possible concurrent read and write happening.

Reply:
https://github.com/mbdavid/LiteDB/pull/2538/files/4bdb4b700d0ed60e72656c84dc674a2c6004682e#r1778334038

Reply:
No. The original issue was actually due to recursive access. That's a difference.

Reply:
I'm talking about https://github.com/mbdavid/LiteDB/issues/2536. It's possible to have multiple threads doing concurrent read (which is fine) but at the same time one thread will be writing to that dictionary, and this situation isn't thread safe.

Reply:
Any access to _buildEntities is guarded by a lock

https://github.com/JKamsker/LiteDB/blob/4bdb4b700d0ed60e72656c84dc674a2c6004682e/LiteDB/Client/Mapper/BsonMapper.cs#L250-L269

Reply:
I'm not talking about `_buildEntities` but `_entities`

Reply:
The double-checked locking has always been there and i cant find any issues concerning this regard. https://github.com/mbdavid/LiteDB/pull/2493/files#diff-db4637fbacd6bcf21b8bc9df379f084f98e4cac70c3d9858bc8df4d28bdbe7efL234

Reply:
Here you go #2563, repro which throws nre